sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -235,9 +235,28 @@ object RapidsBufferCatalog extends Logging with Arm {
    * Lookup the buffer that corresponds to the specified buffer ID and acquire it.
    * NOTE: It is the responsibility of the caller to close the buffer.
    * @param id buffer identifier
+   * @param unspill if true, unspill the buffer to device storage
    * @return buffer that has been acquired
    */
-  def acquireBuffer(id: RapidsBufferId): RapidsBuffer = singleton.acquireBuffer(id)
+  def acquireBuffer(id: RapidsBufferId, unspill: Boolean = false): RapidsBuffer = {
```

Comment:
I think I would prefer to have a completely separate API for the unspill case instead of a boolean flag. It would also let us change the type signature so the return type is not just a `RapidsBuffer`, but one that includes `getColumnarBatch`.  That way we can remove `getColumnarBatch` from `RapidsBuffer` and let the type checker verify that we called the correct API to unspill when we want to convert it to a batch.

Reply:
Agree, I think performing an unspill on `acquireBuffer` is inappropriate.  There may be cases in the future where we want to examine the location of a buffer without triggering a device copy.  However I'm not sure I agree with removing `getColumnarBatch` from `RapidsBuffer`.

IMO the unspill behavior should occur in `getColumnarBatch` since that is the only buffer method that _requires_ the data to be in device memory to satisfy the request.  We could add a separate unspill API if there's a use-case for it, but I would expect most users to only call the existing `getColumnarBatch` to trigger that behavior.  They aren't going to care if the buffer had to unspill to satisfy the request nor want to deal with the logic of doing the unspill themselves before calling `getColumnarBatch`.